File: 2Sum.js
class Solution {
  /**
   * @param {number[]} nums
   * @param {number} target
   * @return {number[]}
   */
  twoSum(nums, target) {
    let seen = {};

    for (const [i, n] of nums.entries()) {
      let diff = target - n;
      if (diff in seen) {
        return [seen[diff], i];
      }
      seen[n] = i;
    }
  }
}

File: 2_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {3: 0}  # val -> index

        for i, n in enumerate(nums):
            difference = target - n
            if difference in seen:
                return [seen[difference], i]
            seen[n] = i
